sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_transport_async.py

```diff
@@ -262,93 +260,41 @@ async def connect(self):
             # are we already connected?
             if self.connected:
                 return
-            await self._connect(self.host, self.port, self.connect_timeout)
-            self._init_socket(self.socket_settings)
+            try:
+            # Building ssl opts here instead of constructor, so that invalid cert error is raised
+            # when client is connecting, rather then during creation. For uamqp exception parity.
+                self.sslopts = self._build_ssl_opts(self.sslopts)
+            except FileNotFoundError as exc:
+            # FileNotFoundError does not have missing filename info, so adding it below.
+            # Assuming that this must be ca_certs, since this is the only file path that
+            # users can pass in (`connection_verify` in the EH/SB clients) through sslopts above.
+            # For uamqp exception parity. Remove later when resolving issue #27128.
+                exc.filename = self.sslopts
+                raise exc
             self.reader, self.writer = await asyncio.open_connection(
-                sock=self.sock,
+                host=self.host,
+                port=self.port,
                 ssl=self.sslopts,
+                family=socket.AF_UNSPEC,
+                proto=SOL_TCP,
                 server_hostname=self.host if self.sslopts else None,
             )
-            # we've sent the banner; signal connect
-            # EINTR, EAGAIN, EWOULDBLOCK would signal that the banner
-            # has _not_ been sent
             self.connected = True
+            sock = self.writer.transport.get_extra_info("socket")
+            if sock:
+                sock.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
```

Comment:
I was able to bring in practically all of the of the keep alive, other socket properties from previous time. These largely apply to Linux & Windows. MacOS doesnt have some of these OS level props, so its ignored. The timeout was not brought over as that makes sockets blocking, which is a no go for async. The idea is to wait wait_for and the timeout param in there to act as the "timeout", which is what we do :)

Reply:
And we're happy that `get_extra_info("socket")` is supported public API? Looks that way, but just confirming (anyway, given that it's Python stdlib, even if it's changed in Python vNext, we'd at least have some warning).

Is there any potential for these calls to generate additional exceptions that might need to be handled on line 289?

Reply:
yeah, that is supported in the public API for all the transports that are supported by streams. I cant believe I missed this [detail](https://docs.python.org/3/library/asyncio-protocol.html#asyncio.BaseTransport.get_extra_info) when I first implemented it. Their example itself shows how to get the underlying socket :)

Looking around I came around its usage in aiohttp where they also set up the KEEP_ALIVE in the same way https://github.com/aio-libs/aiohttp/blob/ffad0878989fc1ea95304406abbf421d627fe366/aiohttp/tcp_helpers.py#L13. 

With that I was fairly confident on using this API to set socket properties, its public and to your point we should have some warning if its going away. 

On line 289, with these options set, any errors raised will be between the `SSLError` and `OSError`. So these will be sufficient :)